x/superfluid/keeper/intermediary_account.go

```diff
@@ -91,7 +100,10 @@ func (k Keeper) GetOrCreateIntermediaryAccount(ctx sdk.Context, denom, valAddr s
 	// create a new account. We use base accounts, as this is what's done for cosmwasm smart contract accounts.
 	// and in the off-chance someone manages to find a bug that forces the account's creation.
 	if !k.ak.HasAccount(ctx, intermediaryAcct.GetAccAddress()) {
-		k.ak.SetAccount(ctx, authtypes.NewBaseAccount(intermediaryAcct.GetAccAddress(), nil, 0, 0))
+		// UNFORKING v2 TODO: I now need to set NextAccountNumber with k.ak.NextAccountNumber(ctx) instead of using zero, due to new invariant checks.
```

Comment:
.

Reply:
So for instance, if I keep this as 0, and run the TestSuperfluidDelegate test, I get the following error trace:

```
            /Users/czar/Documents/osmosis-1/osmosis/x/superfluid/keeper/account.go:61: test panicked: collections: conflict: index uniqueness constrain violation: 0
                goroutine 1423 [running]:
                runtime/debug.Stack()
                	/usr/local/go/src/runtime/debug/stack.go:24 +0x64
                github.com/stretchr/testify/suite.failOnPanic(0x14001430340, {0x1078ec8e0, 0x140019f96c0})
                	/Users/czar/Documents/osmosis-1/osmosis/vendor/github.com/stretchr/testify/suite/suite.go:89 +0x38
                github.com/stretchr/testify/suite.recoverAndFailOnPanic(0x14001430340)
                	/Users/czar/Documents/osmosis-1/osmosis/vendor/github.com/stretchr/testify/suite/suite.go:83 +0x40
                panic({0x1078ec8e0?, 0x140019f96c0?})
                	/usr/local/go/src/runtime/panic.go:770 +0x124
                github.com/cosmos/cosmos-sdk/x/auth/keeper.AccountKeeper.SetAccount({{0x107e0e268, 0x14000b716b0}, {0x107dfdba0, 0x1400148f7b8}, {0x107e5a1b0, 0x140013e2960}, 0x14002486d20, {0x106aba47e, 0x4}, 0x107de3558, ...}, ...)
                	/Users/czar/Documents/osmosis-1/osmosis/vendor/github.com/cosmos/cosmos-sdk/x/auth/keeper/account.go:61 +0x80
                github.com/osmosis-labs/osmosis/v25/x/superfluid/keeper.Keeper.GetOrCreateIntermediaryAccount({{0x107e0e178, 0x14000b714a0}, {{0x107e5a1b0, 0x140013e2960}, 0x14000206a38, {0x107e0e178, 0x14000b71110}, {0x107e0e2b8, 0x14000b71630}, {0x14001716e60, ...}, ...}, ...}, ...)
                	/Users/czar/Documents/osmosis-1/osmosis/x/superfluid/keeper/intermediary_account.go:106 +0x4fc
                github.com/osmosis-labs/osmosis/v25/x/superfluid/keeper.Keeper.SuperfluidDelegate({{0x107e0e178, 0x14000b714a0}, {{0x107e5a1b0, 0x140013e2960}, 0x14000206a38, {0x107e0e178, 0x14000b71110}, {0x107e0e2b8, 0x14000b71630}, {0x14001716e60, ...}, ...}, ...}, ...)
                	/Users/czar/Documents/osmosis-1/osmosis/x/superfluid/keeper/stake.go:241 +0x138
                github.com/osmosis-labs/osmosis/v25/x/superfluid/keeper_test.(*KeeperTestSuite).setupSuperfluidDelegate(0x140015e1808, {0x14001548d40, 0x14, 0x20}, {0x1400265a1e0, 0x14, 0x18}, {0x140022e62a0, 0xb}, 0xf4240)
                	/Users/czar/Documents/osmosis-1/osmosis/x/superfluid/keeper/keeper_test.go:239 +0x4d8
                github.com/osmosis-labs/osmosis/v25/x/superfluid/keeper_test.(*KeeperTestSuite).setupSuperfluidDelegations(0x140015e1808, {0x14001829530, 0x2, 0x1400148f7c0?}, {0x14001c5f800, 0x2, 0x107e48e60?}, {0x140019f8900, 0x2, 0x140010d19e0?})
                	/Users/czar/Documents/osmosis-1/osmosis/x/superfluid/keeper/keeper_test.go:189 +0x3dc
                github.com/osmosis-labs/osmosis/v25/x/superfluid/keeper_test.(*KeeperTestSuite).TestSuperfluidDelegate.func1()
                	/Users/czar/Documents/osmosis-1/osmosis/x/superfluid/keeper/stake_test.go:85 +0x370
```

It seems like the collections module (new to v0.50) ensures that all new accounts have unique account numbers. Which kind of makes sense.